wdl-ast/src/validation.rs

```diff
@@ -32,6 +42,62 @@ impl Diagnostics {
     pub fn add(&mut self, diagnostic: Diagnostic) {
         self.0.push(diagnostic);
     }
+
+    /// Adds a diagnostic to the collection, unless the diagnostic is for an
+    /// element that has an exception for the given rule.
+    ///
+    /// If the diagnostic does not have a rule, the diagnostic is always added.
+    pub fn exceptable_add(
```

Comment:
I didn't want to complicate code in `wdl-ast` where diagnostics can't be excepted, so I decided to leave the `add()` method as it is on `main`.

Instead, I added this `exceptable_add()` method, which requires some additional context/parameters. This method should be used universally in-place of the simple `add()` in `wdl-lint`. But I ran into some issues while making that switch. More in other comments: